File: TravelAgencyRestApi/src/main/java/com/travelagency/travelagencyapplication/dto/response/ActivityResponse.java
package com.travelagency.travelagencyapplication.dto.response;

import com.travelagency.travelagencyapplication.collection.Guide;
import com.travelagency.travelagencyapplication.collection.User;
import com.travelagency.travelagencyapplication.enums.ActivityCostType;
import com.travelagency.travelagencyapplication.enums.ActivityType;
import com.travelagency.travelagencyapplication.enums.PaymentType;
import lombok.*;

import java.time.LocalTime;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ActivityResponse{

    private String id;
    private String title;
    private String destination;
    private Set<String> imagesUrl;
    private String shortDescription;
    private String longDescription;
    private GuideResponse activityGuide;
    private ActivityType activityType;
    private ActivityCostType activityCostType;

    private Double activityAdditionalPrice;
    private LocalTime startAt;
    private LocalTime endAt;

}

File: TravelAgencyFrontend/src/app/types/Activity/ActivityResponse.ts
import {ActivityType} from "./ActivityType";
import {ActivityCostType} from "./ActivityCostType";
import {GuideResponse} from "../Guide/GuideResponse";
import {Validators} from "@angular/forms";

export interface ActivityResponse {
  id?: string;
  title: string;
  destination: string;
  imagesUrl: string[];
  shortDescription: string;
  longDescription: string;
  activityGuide: GuideResponse;
  activityType: ActivityType;
  activityCostType: ActivityCostType;
  activityAdditionalPrice?: number; // Optional field
  startAt:string,
  endAt: string
}
